src/nhp/data/model_data/health_status_adjustment/generate_icb_gams.py

```diff
@@ -0,0 +1,135 @@
+"""Generate GAMs and HSA activity tables"""
+
+import sys
+from functools import reduce
+from typing import Any
+
+import numpy as np
+import pandas as pd
+import pyspark.sql.functions as F
+from pygam import GAM
+from pyspark.sql import DataFrame, SparkSession
+
+from nhp.data.get_spark import get_spark
+from nhp.data.model_data.helpers import create_icb_population_projections
+
+
+def _get_data(spark: SparkSession, save_path: str) -> DataFrame:
+    dfr = (
+        reduce(
+            DataFrame.unionByName,
+            [
+                (
+                    spark.read.parquet(f"{save_path}/{ds}")
+                    .groupBy("fyear", "icb", "age", "sex", "hsagrp")
+                    .count()
+                )
+                for ds in ["ip", "op", "aae"]
+            ],
+        )
+        .filter(~F.col("hsagrp").isin(["birth", "maternity", "paeds", "unknown"]))
+        .filter(F.col("fyear").isin([2023]))
```

Comment:
Should fyear be parameterised rather than fixed in the code?

Reply:
It's fixed in all of the gams code. Might be something to change separately